psi4/src/psi4/lib3index/df_helper.cc

```diff
@@ -1551,282 +1747,303 @@ void DF_Helper::transform_core() {
+                put_tensor(putf, Np, std::make_pair(begin, end), std::make_pair(0, bsize - 1),
+                           std::make_pair(0, wsize - 1), op);
+            }
+        
+        // result is in pQq format
+        } else {
+            
+            // (w|Qb)->(bQw)
+            #pragma omp parallel for num_threads(nthreads_)
+            for (size_t x = 0; x < rblock_size; x++) {
+                for (size_t y = 0; y < bsize; y++) {
+                    for (size_t z = 0; z < wsize; z++) {
+                        Np[y * lblock_size * wsize + (bcount + x) * wsize + z] 
+                            = Fp[z * bsize *rblock_size + x * bsize + y];
+                    }
+                }
+            }
+            timer_off("DFH: (w|Qb)->(bw|Q)");
+            if(!MO_core_){
+                timer_on("DFH: MO to disk");
+                put_tensor(putf, Np, std::make_pair(0, bsize - 1), std::make_pair(begin, end),
+                           std::make_pair(0, wsize - 1), op);
+            }
 
-        std::vector<double> M;
-        std::vector<double> F;
-        M.reserve(total_mem);
-        F.reserve(total_mem);
-        double* Mp = M.data();
-        double* Fp = F.data();
-        double* metp;
+        }
 
-        if (hold_met_) {
-            metp = metric_prep_core(mpower_);
-            for (std::vector<std::string>::iterator itr = order_.begin(); itr != order_.end(); itr++)
-                contract_metric(*itr, metp, Mp, Fp, total_mem);
+    } else {
+        
+        // result is in pqQ format
+        if (std::get<2>(transf_[order_[ind]]) == 2) {  
+            
+            // (w|Qb)->(wbQ)
+            #pragma omp parallel for num_threads(nthreads_)
+            for (size_t z = 0; z < wsize; z++) {
+                for (size_t x = 0; x < rblock_size; x++) {
+                    for (size_t y = 0; y < bsize; y++) {
+                        Np[z * lblock_size * bsize + y * lblock_size + (bcount + x)] 
+                            = Fp[z * rblock_size * bsize + x * bsize + y];
+                    }
+                }
+            }
+            timer_off("DFH: (w|Qb)->(bw|Q)");
+            if(!MO_core_){
+                timer_on("DFH: MO to disk");
+                put_tensor(putf, Np, std::make_pair(0, wsize - 1), std::make_pair(0, bsize - 1),
+                           std::make_pair(begin, end), op);
+            }
 
+        // result is in Qpq format
+        } else if (std::get<2>(transf_[order_[ind]]) == 0) {  
+
+            // (w|Qb)->(Q|wb)
+            #pragma omp parallel for num_threads(nthreads_)
+            for (size_t x = 0; x < rblock_size; x++) {
+                for (size_t z = 0; z < wsize; z++) {
+                    #pragma omp simd
+                    for (size_t y = 0; y < bsize; y++) {
+                        Np[(bcount + x) * wsize * bsize + z * bsize + y] 
+                            = Fp[z * rblock_size * bsize + x * bsize + y];
+                    }
```

Comment:
Any reason to use SIMD here and C_DCOPY below?

Reply:
No reason other than inconsistency.  I will revert to C_DCOPY.